packages/ember-truth-helpers/src/helpers/and.ts

```diff
@@ -1,19 +1,11 @@
-import { helper } from '@ember/component/helper';
 import truthConvert from '../utils/truth-convert.ts';
 import type { MaybeTruth } from '../utils/truth-convert.ts';
 
-export interface AndSignature {
-  Args: {
-    Positional: MaybeTruth[];
-  };
-  Return: boolean;
-}
-
-export default helper<AndSignature>((params) => {
+export default function and<T extends MaybeTruth[]>(...params: [...T]) {
```

Comment:
@NullVoxPopuli using spread causes an issue - today, arguments gets lazily evaluated and spread makes it process eager.

e.g. imagine use case

```hbs
{{and (foo "A") (bar "B")}}
```

today, if `(foo "A")` returns `false`, then `(bar "B")` does not get evaluated.
but this line change makes it evaluated which is definitely not what we want

Reply:
that's just how plain functions work though: https://github.com/glimmerjs/glimmer-vm/blob/68d371bdccb41bc239b8f70d832e956ce6c349d8/packages/%40glimmer/manager/lib/internal/defaults.ts#L38

Reply:
to make `or` lazy or short-circuit, it needs to be built in to the VM.

Reply:
I wonder if `or` and `and` need to remain "classic" I wonder if that means converting to class helpers so that we can correctly use generics?